src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -2031,7 +2052,12 @@ public void onFailure(Throwable t) {
                 persistOk=false;
                 LOGGER.log(Level.WARNING, "Error persisting FlowNode storage before shutdown", ioe);
             }
-            persistedClean = persistOk;
+            if (Jenkins.get().isTerminating()) {
+                // Only modify persistedClean if Jenkins is shutting down.
+                // TODO: We could set persistedClean to true when the build is just being paused, but then we would need
+                // to null it out and save again when the build was unpaused.
```

Comment:
Looks like `FlowDurabilityTest.testPauseForcesLowDurabilityToPersist` is failing because it expects that pausing the build will make it be able to resume, so I will need to think about this.

Reply:
Addressed by https://github.com/jenkinsci/workflow-cps-plugin/pull/363/commits/2214c421c47c01e84a2c7218912f0587799f8252.